packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -0,0 +1,229 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { JsonCompatibleReadOnly } from "../../../change-family";
+import {
+    FieldChangeEncoder,
+    FieldChangeHandler,
+    FieldChangeMap,
+    FieldChangeRebaser,
+    FieldKind,
+    Multiplicity,
+    ModularChangeFamily,
+    FieldKinds,
+} from "../../../feature-libraries";
+import { FieldKindIdentifier } from "../../../schema-stored";
+import { Delta, FieldKey } from "../../../tree";
+import { brand } from "../../../util";
+
+type ValueChangeset = FieldKinds.ReplaceOp<number>;
+
+const valueHandler: FieldChangeHandler<ValueChangeset> = {
+    rebaser: FieldKinds.replaceRebaser(),
+    encoder: new FieldKinds.ValueEncoder<ValueChangeset & JsonCompatibleReadOnly>(),
+
+    intoDelta: (change, deltaFromChild) => [{
+        type: Delta.MarkType.Modify,
+        setValue: (change as FieldKinds.Replacement<number>).new,
+    }],
+};
+
+const valueField = new FieldKind(
+    brand("Value"),
+    Multiplicity.Value,
+    valueHandler,
+    (a, b) => false,
+    new Set(),
+);
+
+const singleNodeEncoder: FieldChangeEncoder<FieldChangeMap> = {
+    encodeForJson: (formatVersion, change, encodeChild) => encodeChild(change),
+    decodeJson: (formatVersion, change, decodeChild) => decodeChild(change),
+};
+
+const singleNodeRebaser: FieldChangeRebaser<FieldChangeMap> = {
+    compose: (changes, composeChild) => composeChild(...changes),
```

Comment:
```suggestion
    compose: (changes, composeChild) => composeChild(changes),
```

Reply:
This suggestion doesn't currently work. Maybe it is unexpected that the signatures of `compose` and `composeChild` are different. Do you think we should change `composeChild` to not use a rest parameter?

Reply:
Yes, we generally try to stay away from rest parameters unless there's a strong case for them. They have an O(n) time cost compared to arrays and they make de debugging experience a little worse.